File: Stock buy and sell II[5-July-2023]/Solution.py
#  >> PYTHON CODE <<

class Solution:
    def stockBuyAndSell(self, n : int, prices : List[int]) -> int:
        buy=-1
        profit=0
        
        for i in range(n):
            if(buy==-1):
                buy=prices[i]
                
            if i<n-1 and prices[i]>prices[i+1]:
                profit+=(prices[i]-buy)
                buy=-1
            elif buy!=-1 and i==n-1:
                profit+=(prices[i]-buy)
        
        return profit

File: Stock buy and sell II[5-July-2023]/Solution.java
//  >> JAVA CODE <<

class Solution {
    public static int stockBuyAndSell(int n, int[] prices) {
        int buy = -1;
        int profit = 0;
    
        for (int i = 0; i < n; i++) {
            if (buy == -1) {
                buy = prices[i];
            }
    
            if (i < n - 1 && prices[i] > prices[i + 1]) {
                profit += (prices[i] - buy);
                buy = -1;
            } else if (buy != -1 && i == n - 1) {
                profit += (prices[i] - buy);
            }
        }
    
        return profit;

    }
}